### ISSUE TITLE
camelCase labels and properties

### ISSUE THREAD

**Issue by SyntaxUnknown** [ExRam/ExRam.Gremlinq#8]
I'm getting some push back around how the library is storing properties in the same casing as the C# file (PascalCase). I see that you added a .WithLowercase option for the model, but that's only addressing labels and it's exactly what I need.

So, using that as the inspiration, I added "WithCamelcase":

```
var graph = g
                .WithLogger(_logger)
                .WithModel(GraphModel.Dynamic()
                .WithCamelcase())
                .WithExecutor(...)     
```

It seems to work well in my tests. Serialization works without having to add any special attributes to your C# Vertex/Edge derived classes. Where clauses also seem to work just fine.

Internally, I created a CamelcaseGraphModel, which is is similar to LowercaseGraphModel except it calls a string extension method I created called "ToCamelCase":

```
public Option<string> TryGetConstructiveLabel(Type elementType) => _baseModel.TryGetConstructiveLabel(elementType).Map(x => x.ToCamelCase());

public Option<string[]> TryGetFilterLabels(Type elementType) => _baseModel.TryGetFilterLabels(elementType).Map(x => x.Select(y => y.ToCamelCase()).ToArray());
```
The only 'hacky bit' in the static GetIdentifier method:

```
internal static object GetIdentifier(this IGraphModel model, Type elementType, string memberName)
```
Since it's static, it's hard to extend, so for now, I just added:

```
            if (model is CamelcaseGraphModel)
            {
                return memberName.ToCamelCase();
            }

            return memberName;
```
The only other thing I can think of here is to enhance the IGraphModel interface to allow us to specify some sort of optional transformation method, but I'm guessing you want to keep that interface clean.

Let me know what you think. I'll be happy to submit a PR.

**Comment by danielcweber** [comment]
Agreed, name lookup for members should be a thing on the I(Graph, Vertex, Edge)Model interfaces.

**Comment by danielcweber** [comment]
Fixed by #10.
